File: app/src/main/java/org/kindone/willingtodo/pomodorotimer/TimerInitial.java
package org.kindone.willingtodo.pomodorotimer;

/**
 * Created by kindone on 2017. 1. 1..
 */

public class TimerInitial extends TimerState {

    public TimerInitial(TimerContext context) {
        super(context);
    }

    @Override
    public boolean start() {
        changeState(new TimerRunning(mContext, mContext.getDurationMs()));
        return CHANGED;
    }

    @Override
    public boolean stop() {
        // do nothing
        return UNCHANGED;
    }

    @Override
    public boolean pause() {
        // do nothing
        return UNCHANGED;
    }

    @Override
    public boolean resume() {
        // do nothing
        return UNCHANGED;
    }

    @Override
    public long getRemainingTimeMs() {
        return mContext.getDurationMs();
    }

    public boolean isRunning() {
        return false;
    }

    public boolean isStopped() {
        return false;
    }
}

File: app/src/main/java/org/kindone/willingtodo/pomodorotimer/TimerInitial.kt
package org.kindone.willingtodo.pomodorotimer

/**
 * Created by kindone on 2017. 1. 1..
 */

class TimerInitial(context: TimerContext) : TimerState(context) {

    override fun start(): Boolean {
        changeState(TimerRunning(mContext, mContext.getDurationMs()))
        return TimerState.CHANGED
    }

    override fun stop(): Boolean {
        // do nothing
        return TimerState.UNCHANGED
    }

    override fun pause(): Boolean {
        // do nothing
        return TimerState.UNCHANGED
    }

    override fun resume(): Boolean {
        // do nothing
        return TimerState.UNCHANGED
    }

    override fun getRemainingTimeMs(): Long {
        return mContext.getDurationMs()
    }

    override fun isRunning(): Boolean {
        return false
    }

    override fun getStartedTimeMs(): Long {
        return -1L
    }

    override fun isStopped(): Boolean {
        return false
    }
}
